File: flutterapp/lib/models/movie_model.dart
import 'dart:convert';

class MovieItem {
  final String id;
  final String createdAt;
  final String movieId;
  final String movieUrl;
  final String name;
  final String topic;
  final String updatedAt;

  MovieItem({
    required this.id,
    required this.createdAt,
    required this.movieId,
    required this.movieUrl,
    required this.name,
    required this.topic,
    required this.updatedAt,
  });

  factory MovieItem.fromJson(Map<String, dynamic> json) {
    return MovieItem(
      id: json['_id'],
      createdAt: json['created_at'],
      movieId: json['movie_id'],
      movieUrl: json['movie_url'],
      name: json['name'],
      topic: json['topic'],
      updatedAt: json['updated_at'],
    );
  }
}

class MoviesResponse {
  final List<MovieItem> movieItems;
  final int totalCount;

  MoviesResponse({
    required this.movieItems,
    required this.totalCount,
  });

  factory MoviesResponse.fromJson(Map<String, dynamic> json) {
    var list = json['movie_items'] as List;
    List<MovieItem> movieItemsList = list.map((i) => MovieItem.fromJson(i)).toList();

    return MoviesResponse(
      movieItems: movieItemsList,
      totalCount: json['total_count'],
    );
  }
}

File: goapp/models/movieModel.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Movie struct {
	Id         primitive.ObjectID `bson:"_id"`
	Name       *string            `json:"name" validate:"required"`
	Topic      *string            `json:"topic" validate:"required"`
	Movie_URL  *string            `json:"movie_url" validate:"required"`
	Created_at time.Time          `json:"created_at"`
	Updated_at time.Time          `json:"updated_at"`
	Movie_id	interface{}				  `json:"movie_id"`

}